flask_project/campaign_manager/test/test_upload_coverage.py

```diff
@@ -0,0 +1,71 @@
+# coding=utf-8
+import unittest
+from unittest import mock
+from campaign_manager.insights_functions.upload_coverage import UploadCoverage
+from campaign_manager.test.helpers import CampaignObjectTest
+
+
+class UploadCoverageTestCase(unittest.TestCase):
+    """Test the insights function to upload shapefile for creating coverage."""
+
+    def setUp(self):
+        """Constructor."""
+        self.campaign = CampaignObjectTest()
+        self.uuid = self.campaign.uuid
+        self.upload_coverage = UploadCoverage(campaign=self.campaign)
+
+    def tearDown(self):
+        """Destructor."""
+        pass
+
+    def test_initiate(self):
+        self.UploadCoverageTest = \
+            self.upload_coverage.initiate(additional_data={})
+        self.assertIsNotNone(self, self.UploadCoverageTest)
+
+    def test_get_ui_html_file(self):
+        ui_html = self.upload_coverage.get_ui_html_file()
+        self.assertEquals(ui_html, 'upload_coverage')
+
+    def test_get_summary_html_file(self):
+        self.upload_coverage.get_summary_html_file = mock.MagicMock(return_value='html summary test')
+        summary_result = self.upload_coverage.get_summary_html_file
+        self.assertIsNotNone(summary_result)
+        self.assertEquals(summary_result.return_value, 'html summary test')
+
+    def test_details_html_file(self):
+        self.upload_coverage.get_details_html_file = mock.MagicMock(return_value='html details test')
+        details_result = self.upload_coverage.get_details_html_file
+        self.assertIsNotNone(details_result)
+        self.assertEquals(details_result.return_value, 'html details test')
+
+    def test_get_data_from_provider(self):
+        shapefile_file = self.upload_coverage.get_data_from_provider()
+        self.assertIsNotNone(shapefile_file)
+        self.assertEquals(shapefile_file['type'], 'FeatureCollection')
+
+    def test_process_data(self):
+        process_data = UploadCoverage.process_data(
+            self, raw_datas={('data1', 1234), ('data2', 2345)})
+        self.assertEquals(process_data, {('data1', 1234), ('data2', 2345)})
+
+    def test_get_coverage_files(self):
+        output_files = self.upload_coverage.get_coverage_files()
+        expected_output_files = \
```

Comment:
you don't need self.uuid, because of campaign has uuid
please use self.campaign.uui intead, because it is owning by campaign